src/Core/Router.php

```diff
@@ -2,21 +2,25 @@
 
 namespace TrendingRepos\Core;
 
+use TrendingRepos\App;
 use TrendingRepos\Exception\RouterException;
 
 class Router 
 {
     public $routes;
     public $request;
+    public $app;
 
     public function __construct(array $routes, Request $request)
     {
         $this->routes = $routes;
         $this->request = $request;
     }
 
-    public function get() 
+    public function get(App $app) 
```

Comment:
Can you please explain why did you decide that this method is the best place to inject something to `Router`? 🤔

Reply:
You told me to use only one `App` instance which is already done in `index.php` so ..
in `App.php` i tried to pass an instance of it to the router and i call `$this->router->get()` inside App so i find this is the only place to pass it there.

Now i'm not sure how to pass an app instance to all controllers ! 
i think of building a constrcutor in a parent `Controller` class where all sub controllers will extend `$this->app->registry` but this will require to make a new App instance !!